src/Spatial/Euclidean/ILine2D.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace MathNet.Spatial.Euclidean
+{
+    /// <summary>
+    /// This interface represents a line in cartesian space
+    /// </summary>
+    public interface ILine2D : IEquatable<ILine2D>
+    {
+        /// <summary>
+        /// Returns the points at which the line crosses on the x axis
+        /// </summary>
+        /// <param name="x">An optional x coordinate</param>
+        /// <returns>A list of crossing points</returns>
+        Point2D[] XIntercept(double x = 0);
```

Comment:
Don't return an array. `IEnumerable<Point2D>` is preferable because it can be precomputed or cached or calculated lazily or eagerly.

Reply:
this is what I had ... Johan asked to change it...